src/python/pants/fs/archive.py

```diff
@@ -67,7 +69,17 @@ class TarArchiver(Archiver):
 
   def _extract(self, path_or_file, outdir, **kwargs):
     with open_tar(path_or_file, errorlevel=1, **kwargs) as tar:
-      tar.extractall(outdir)
+      if PY2:
+        # TarFile in Py2 grabs files as byte strings and doesn't store the encoding.
+        # Unicode will thus fail, as it tries to coerce it through Ascii.
+        # See https://superuser.com/questions/60379/how-can-i-create-a-zip-tgz-in-linux-such-that-windows-has-proper-filenames/190786#190786.
+        decoded_members = []
+        for m in tar.getmembers():
+          m.name = m.name.decode('utf-8')
+          decoded_members.append(m)
+        tar.extractall(outdir, members=decoded_members)
+      else:
+        tar.extractall(outdir)
```

Comment:
This change is necessary because of removing `suffix=b''` in `temporary_dir()`, which means `temporary_dir` will return a unicode file name instead of bytes—a good thing and necessary for Py3 to work. 

In Py2, TarFile manges unicode in file names. See https://superuser.com/questions/60379/how-can-i-create-a-zip-tgz-in-linux-such-that-windows-has-proper-filenames/190786#190786 for more information and the basis of this fix.